pkg/providers/vsphere/vsphere.go

```diff
@@ -1371,23 +1371,23 @@ func (p *vsphereProvider) GetInfrastructureBundle(clusterSpec *cluster.Spec) *ty
 }
 
 func (p *vsphereProvider) DatacenterConfig() providers.DatacenterConfig {
-	return p.datacenterConfig
+	return p.datacenterConfig.ConvertConfigToConfigGenerateStruct()
 }
 
 func (p *vsphereProvider) MachineConfigs() []providers.MachineConfig {
 	var configs []providers.MachineConfig
 	controlPlaneMachineName := p.clusterConfig.Spec.ControlPlaneConfiguration.MachineGroupRef.Name
 	workerMachineName := p.clusterConfig.Spec.WorkerNodeGroupConfigurations[0].MachineGroupRef.Name
 	p.machineConfigs[controlPlaneMachineName].Annotations = map[string]string{p.clusterConfig.ControlPlaneAnnotation(): "true"}
-	configs = append(configs, p.machineConfigs[controlPlaneMachineName])
+	configs = append(configs, p.machineConfigs[controlPlaneMachineName].ConvertConfigToConfigGenerateStruct())
```

Comment:
@danbudris will the change here affect your PR on MachineConfigs? we remove the status field.

Reply:
so in my PR [I'm adding the `OSFamily()` method to the `MachineConfig` interface](https://github.com/aws/eks-anywhere/pull/286/files#diff-8f631100c024f9035d740ab95c024634e695f200be9fde670b7a85a101d1b8edR51)

along with the [corresponding implementation in the `VSphereMachineConfig`](https://github.com/aws/eks-anywhere/pull/286/files#diff-88ba4e36f3f46665ba8a2a980719c15dca7167959b8bd04d787184402f39c602R52)

Reply:
so as long as the MachineConfigs being returned still implement the `MachineConfig` interface in the `Provider` package, we're all good -- I don't need access to status or kind even

Reply:
roger. thanks!